File: algorithm_study_python/baekjoon_online/p_1003/p_1003.py
'''
Input:
41
0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40

Input2:
1
3


'''

MAX_FIBONACCI = 41
# fibonacci = [0 for i in range(MAX_FIBONACCI)]
fibonacci = [0] * MAX_FIBONACCI
fibonacci[1] = 1
for i in range(2, MAX_FIBONACCI) :
    fibonacci[i] = fibonacci[i - 1] + fibonacci[i - 2]

count = int(input())

for i in range(count) :
    item = int(input())

    if item == 0 :
        a = 1
        b = 0
    else :
        a = fibonacci[item - 1]
        b = fibonacci[item]
    print(a, b)

File: algorithm_study_cpp/algorithm_study_cpp/p_1003.cpp
//
//  p_1003.cpp
//  algorithm_study_cpp
//
//  Created by Dukwon Nam on 14/11/2018.
//  Copyright © 2018 Dukwon Nam. All rights reserved.
//

#include <stdio.h>
#include <iostream>
#include "p_1003.hpp"

using namespace std;

# define MAX_FIBONACCI 41

/*
INPUT
3
0
1
3

 */

int main_1003() {
    int count, item;
    int a, b;
    int i;
    
    int fibonacci[MAX_FIBONACCI];
    fibonacci[0] = 0;
    fibonacci[1] = 1;
    for (i = 2; i < MAX_FIBONACCI; i++) {
        fibonacci[i] = fibonacci[i - 1] + fibonacci[i - 2];
    }
    
    // First line
    cin >> count;
    
    for (; count > 0; count--) {
        cin >> item;
        
        if (item == 0) {
            a = 1;
            b = 0;
        } else {
            a = fibonacci[item - 1];
            b = fibonacci[item];
        }
        cout << a << " " << b << endl;
    }
    
    return 0;
}
